sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreSuite.scala

```diff
@@ -343,6 +343,63 @@ class RocksDBStateStoreSuite extends StateStoreSuiteBase[RocksDBStateStoreProvid
     }
   }
 
+  Seq(true, false).foreach { colFamiliesEnabled =>
+    test(s"rocksdb range scan - variable size non-ordering columns with non-zero start ordinal " +
+      s"with colFamiliesEnabled=$colFamiliesEnabled") {
+
+      tryWithProviderResource(newStoreProvider(keySchema,
+        RangeKeyScanStateEncoderSpec(
+          keySchema, Seq(1)), colFamiliesEnabled)) { provider =>
+        val store = provider.getStore(0)
+
+        // use non-default col family if column families are enabled
+        val cfName = if (colFamiliesEnabled) "testColFamily" else "default"
+        if (colFamiliesEnabled) {
+          store.createColFamilyIfAbsent(cfName,
+            keySchema, valueSchema,
+            RangeKeyScanStateEncoderSpec(keySchema, Seq(1)))
+        }
+
+        val timerTimestamps = Seq(931L, 8000L, 452300L, 4200L, -1L, 90L, 1L, 2L, 8L,
```

Comment:
nit: Is it intentional to use Long constants here while we cast to Int again?

Reply:
Done